userspace/libsinsp/threadinfo.cpp

```diff
@@ -976,7 +976,8 @@ sinsp_threadinfo* sinsp_threadinfo::get_main_thread()
 		}
 	}
 
-	return &*m_main_thread;
+	auto ret = m_main_thread.lock();
+	return &(*ret);
```

Comment:
What's the lock for here? Is it safe to return the result? (are we returning with some sort of mutex held? where is it unlocked?)

Reply:
m_main_thread is a weak_ptr. lock() is not a mutex - it tests whether there's a valid object that the weak_ptr references. If there's a valid object, lock() returns that object.

Reply:
ok, got it